client/src/plugins/settings/SettingsForm.js

```diff
@@ -156,3 +162,66 @@ function sortSchemaByOrder(schema) {
 
   return sortedObj;
 }
+
+/**
+ * Resolves a path relative to the current path or itself if its an absolute path (contains a dot).
+ *
+ * @param {string} currentPath
+ * @param {string} targetPath - Path to resolve. If it contains a dot, it is considered an absolute path and returned as is.
+ * @returns {string} The resolved path
+ */
+export function resolvePath(currentPath, targetPath) {
+  if (targetPath.includes('.')) {
+    return targetPath;
+  }
+
+  const currentSegments = currentPath.split('.');
+
+  currentSegments.pop();
+
+  const resolvedSegments = [ ...currentSegments, targetPath ];
+  return resolvedSegments.join('.');
+}
+
+/**
+ * @typedef {Object} Condition A condition to be met for a form field to be shown
+ * @property {string} [property] - The key of the property to check, can be relative.
+ * @property {Array<Condition>} [allMatch] - A list of child conditions that must all be met.
+ * @property {any} [equals] - The value the property should be equal to.
+ * @property {Array<any>} [oneOf] - A list of possible values for the property.
+ *
+ * @example
+ * { property: 'some.key', equals: true }
+ * { property: 'some.key', oneOf: [ 'a', 'b', 'c' ] }
+ * { allMatch: [ { property: 'some.key', equals: true }, { property: 'other.key', oneOf: [ 'a', 'b' ] } ] }
+ */
+
+/**
+ * Checks if a given condition is met based on the provided form values.
+ *
+ * @param {string} propName - The base path for the property being evaluated.
+ * @param {Object} values - The object containing all form values, used to cross reference other section/fields
+ * @param {Condition} condition - The condition object to evaluate.
+ * @returns {boolean} True if the condition is met, false otherwise.
+ */
+export function isConditionMet(propName, values, condition) {
+  if (condition.allMatch) {
+    return condition.allMatch.every((childCondition) => isConditionMet(propName, values, childCondition));
+  }
+
+  if (!condition.property) {
+    return false;
+  }
+  const conditionPropPath = resolvePath(propName, condition.property);
+  const conditionPropValue = getIn(values, conditionPropPath);
+
+  if ('equals' in condition) {
```

Comment:
At first it looked inconsistent but I understand the logic (e.g. `{ equals: true }`.